File: src/catalog-api/Catalog.Domain/Entities/Product.cs
namespace Catalog.Domain.Entities
{
    public class Product : DomainEntity
    {                
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public string PictureFileName { get; set; }
        public Guid ProductTypeId { get; set; }
        public ProductType ProductType { get; set; }
        public Guid ProductBrandId { get; set; }
        public ProductBrand ProductBrand { get; set; }        
        public int AvailableStock { get; set; }
    }
}

File: src/store-app/src/app/feature/catalog/shared/model/product.ts
export interface Product {
  id: string;
  name: string;
  description: string;
  price: number;
  pictureFileName: string;
  productTypeId: string; // Assuming GUID as string for simplicity
  productType: ProductType;
  productBrandId: string; // Assuming GUID as string for simplicity
  productBrand: ProductBrand;
  availableStock: number;
}

interface ProductType {
  id: string;
  type: string;
}

interface ProductBrand {
  id: string;
  brand: string;
}
